src/payment/strategies/masterpass/masterpass-payment-strategy.spec.ts

```diff
@@ -0,0 +1,257 @@
+import { createClient as createPaymentClient } from '@bigcommerce/bigpay-client';
+import { Action } from '@bigcommerce/data-store';
+import createAction from '@bigcommerce/data-store/lib/create-action';
+import { createRequestSender, RequestSender } from '@bigcommerce/request-sender';
+import { createScriptLoader } from '@bigcommerce/script-loader';
+import { Observable } from 'rxjs';
+
+import {
+    PaymentActionCreator, PaymentInitializeOptions,
+    PaymentMethod,
+    PaymentMethodActionCreator, PaymentMethodActionType,
+    PaymentMethodRequestSender, PaymentRequestSender
+} from '../../';
+import { getCartState } from '../../../cart/carts.mock';
+import { createCheckoutStore, CheckoutRequestSender, CheckoutStore, CheckoutValidator } from '../../../checkout';
+import { getCheckoutState } from '../../../checkout/checkouts.mock';
+import { MissingDataError } from '../../../common/error/errors';
+import { getConfigState } from '../../../config/configs.mock';
+import { getCustomerState } from '../../../customer/customers.mock';
+import { OrderActionCreator, OrderActionType, OrderRequestBody, OrderRequestSender } from '../../../order';
+import { PaymentActionType } from '../../payment-actions';
+import { getMasterpass, getPaymentMethodsState, getStripe } from '../../payment-methods.mock';
+
+import { MasterpassPaymentStrategy, MasterpassScriptLoader } from './';
+import { Masterpass } from './masterpass';
+import { getMasterpassScriptMock } from './masterpass.mock';
+
+describe('MasterpassPaymentStragegy', () => {
+    let strategy: MasterpassPaymentStrategy;
+
+    let orderRequestSender: OrderRequestSender;
+    let requestSender: RequestSender;
+
+    let store: CheckoutStore;
+    let orderActionCreator: OrderActionCreator;
+    let paymentActionCreator: PaymentActionCreator;
+    let paymentMethodActionCreator: PaymentMethodActionCreator;
+    let scriptLoader: MasterpassScriptLoader;
+
+    let initOptions: PaymentInitializeOptions;
+    let paymentMethodMock: PaymentMethod;
+    let stripePaymentMethodMock: PaymentMethod;
+    let masterpassScript: Masterpass;
+
+    beforeEach(() => {
+        initOptions = {
+            methodId: 'masterpass',
```

Comment:
You are missing the 'masterpass' property of the Initialize options 🍹

Reply:
Well, you actually need it, unless you want the strategy to throw an exception at line 46

Reply:
You're right, I was still working on the tests